src/main/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -82,9 +82,15 @@
     abstract public SamReader open(final File file);
 
     public SamReader open(final Path path) {
-        final SamInputResource r = SamInputResource.of(path, getPathWrapper());
+        return open(path, getPathWrapper(), Function.identity());
+    }
+
+    public SamReader open(final Path path,
+            Function<SeekableByteChannel, SeekableByteChannel> pathWrapper,
+            Function<SeekableByteChannel, SeekableByteChannel> indexWrapper) {
```

Comment:
It's not problematic to share wrappers (they have no state). Those files are accessed differently, so it would be a great coincidence if the best caching&prefetching for one were also the best for the other.